### PR TITLE
Fix cbin_file_path

### LINKED ISSUE #3523: CompressedBinaryIblExtractor does not save cbin_file relative
Hi all,

i notices when dumping a CompressedBinaryIblExtractor to json the path to cbin_file is not saved relative but absolute, which leads to problems when processing it elsewhere.

```
"class": "spikeinterface.extractors.cbin_ibl.CompressedBinaryIblExtractor",
"module": "spikeinterface",
"version": "0.101.2",
"kwargs": {
    "folder_path": "pathremoved/20241105_r0118_wt_1456/ephys",
    "load_sync_channel": false,
    "cbin_file": "/mnt/pathremoved/20241105_r0118_wt_1456/ephys/20241105_r0118_wt_1456.imec0.ap.cbin"
},
...
 "relative_paths": true
```

I found the culprit : 
https://github.com/SpikeInterface/spikeinterface/blob/e525d85948009e7dc57579a8a428502a5534c111/src/spikeinterface/core/core_tools.py#L338
assumes paths should have "path" in the name to be made relative. This should also include "cbin_file" or maybe just "file".

Best,
Artur

### PR DESCRIPTION
**Pull request by alejoe91**
Fixes #3523

### CODE REVIEW

File: src/spikeinterface/extractors/cbin_ibl.py [hunk 1]
```diff
@@ -41,14 +44,21 @@ class CompressedBinaryIblExtractor(BaseRecording):
...
         from neo.rawio.spikeglxrawio import read_meta_file
 
         try:
             import mtscomp
         except ImportError:
             raise ImportError(self.installation_mesg)
-        if cbin_file is None:
+        if cbin_file is not None:
+            warnings.warn(
+                "The `cbin_file` argument is deprecated, please use `cbin_file_path` instead", DeprecationWarning
```
Comment (alejoe91) [inline src/spikeinterface/extractors/cbin_ibl.py:58]:
```suggestion
                "The `cbin_file` argument is deprecated, please use `cbin_file_path` instead", DeprecationWarning, stacklevel=2
```

[the remark this answers is not in the document]
Comment (alejoe91) [reply src/spikeinterface/extractors/cbin_ibl.py]: good catch

File: src/spikeinterface/extractors/cbin_ibl.py [hunk 2]
```diff
@@ -119,7 +131,7 @@ def __init__(self, folder_path=None, load_sync_channel=False, stream_name="ap",
         self._kwargs = {
             "folder_path": str(Path(folder_path).resolve()),
             "load_sync_channel": load_sync_channel,
-            "cbin_file": str(Path(cbin_file).resolve()),
+            "cbin_file_path": str(Path(cbin_file_path).resolve()),
```
[the remark this answers is not in the document]
Comment (alejoe91) [reply src/spikeinterface/extractors/cbin_ibl.py:134]: No this is back compatible since we kept both arguments

**Review by zm711 — APPROVED** [review]
I wonder if it would be nice for provenance to explain why this PR worked. The changes here by themselves don't make sense without doing that the tools scan for "path" in the name. I guess in the future people can read my comment and the issue.

But as the user said this fixes it.

File: src/spikeinterface/extractors/cbin_ibl.py [hunk 3]
```diff
@@ -30,8 +31,10 @@ class CompressedBinaryIblExtractor(BaseRecording):
     stream_name : {"ap", "lp"}, default: "ap".
         Whether to load AP or LFP band, one
         of "ap" or "lp".
-    cbin_file : str or None, default None
+    cbin_file_path : str or None, default None
```
Comment (h-mayorquin) [inline src/spikeinterface/extractors/cbin_ibl.py:34]:
```suggestion
    cbin_file_path : str, Path or None, default None
```

**Review by h-mayorquin — APPROVED** [review]
LGTM. 

I would suggest adding the date on the deprecation warnings so we eventually remove them. and a minor typing thing on the docstring.

**Pull request merged**